### PR TITLE
Catch exception thrown by the RestoreTask and log at the verbose level

### PR DESCRIPTION
**Pull request by joelverhagen**
Done:

1. Catch exception thrown by the `RestoreTask` and log at the verbose level
1. In places where we have a custom retry loop, do not use the loop in `HttpRetryHandler`
1. When we throw a `FatalProtocolException`, do not log an error (since the caller should log)
1. Restore all projects before running .xproj tests

Fix https://github.com/NuGet/Home/issues/3672. The ask here was to fail quicker when there is a bad source. I found that we're trying 9 times... since there are two retry loops with 3 tries each. Also, I trimmed out some duplicate logging.

/cc @emgarten @alpaix

### CODE REVIEW

File: src/NuGet.Core/NuGet.Build.Tasks/RestoreTask.cs [hunk 1]
```diff
@@ -68,10 +69,10 @@ public class RestoreTask : Task
 
         public override bool Execute()
         {
-            if (RestoreGraphItems.Length < 1)
+            System.Console.WriteLine($"Process ID: {System.Diagnostics.Process.GetCurrentProcess().Id}");
```
Comment (alpaix) [inline src/NuGet.Core/NuGet.Build.Tasks/RestoreTask.cs]: Remove?

File: src/NuGet.Core/NuGet.Build.Tasks/RestoreTask.cs [hunk 2]
```diff
@@ -87,6 +88,37 @@ public override bool Execute()
...
+
+                return false;
+            }
+        }
+
+        private async Task<bool> ExecuteAsync(Common.ILogger log)
+        {
+            if (RestoreGraphItems.Length < 1)
+            {
+                log.LogWarning("Unable to find a project to restore!");
```
Comment (alpaix) [inline src/NuGet.Core/NuGet.Build.Tasks/RestoreTask.cs]: Should be localized

**Review by emgarten — APPROVED** [review]

File: src/NuGet.Core/NuGet.Build.Tasks/RestoreTask.cs (hunk 2 above)
Comment (joelverhagen) [reply src/NuGet.Core/NuGet.Build.Tasks/RestoreTask.cs]: Fixed.

**Review by alpaix — APPROVED** [review]
LGTM

**Pull request merged**